libraries/Microsoft.Bot.Builder/AutoSaveStateMiddleware.cs

```diff
@@ -12,13 +12,13 @@ namespace Microsoft.Bot.Builder
     ///  Middleware that will call `read()` and `write()` in parallel on multiple `BotState`
     ///  instances.
     /// </summary>
-    public class BotStateSet : IMiddleware
+    public class AutoSaveStateMiddleware : IMiddleware
     {
         /// <summary>
-        /// Initializes a new instance of the <see cref="BotStateSet"/> class.
+        /// Initializes a new instance of the <see cref="AutoSaveStateMiddleware"/> class.
         /// </summary>
         /// <param name="botStates">initial list of <see cref="BotState"/> objects to manage.</param>
-        public BotStateSet(params BotState[] botStates)
+        public AutoSaveStateMiddleware(params BotState[] botStates)
         {
             BotStates.AddRange(botStates);
```

Comment:
Please add a null check to this. The BotStates is just a list, and will happily accept nulls. We should throw if nothing is passed.